### PR TITLE
XW-1211 | track specific article click events

### PR DESCRIPTION
**Pull request by Warkot**
## Links

* http://wikia-inc.atlassian.net/browse/XW-1211

## Description

A specific click tracking.

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: front/main/app/components/article-content.js [hunk 1]
```diff
@@ -117,6 +119,14 @@ export default Ember.Component.extend(
 			this.destroyChildComponents();
 		},
 
+		click() {
+			const label = this.getTrackingEventLabel(event);
+
```
Comment (hakubo) [inline front/main/app/components/article-content.js]: Missed parameter? Also you could use destructuring here {target}

File: front/main/app/components/article-content.js [hunk 2]
```diff
@@ -176,6 +186,26 @@ export default Ember.Component.extend(
 			}
 		},
 
+		/**
+		 * @param event
+		 * @returns {string}
+		 */
+		getTrackingEventLabel(event) {
```
Comment (hakubo) [inline front/main/app/components/article-content.js]: This only cares about Dom element, you could pass here only that

File: front/main/app/components/article-content.js [hunk 3]
```diff
@@ -176,6 +187,24 @@ export default Ember.Component.extend(
 			}
 		},
 
+		/**
+		 * @param {jQuery} $element — element of which context is to be checked
+		 * @returns {string}
+		 */
+		getTrackingEventLabel($element) {
+			if ($element.length) {
```
Comment (tomasznapieralski) [inline front/main/app/components/article-content.js]: what if `$element` is `undefined`?

File: front/main/app/components/article-content.js [hunk 4]
```diff
@@ -176,6 +187,24 @@ export default Ember.Component.extend(
...
+				if ($element.closest('.portable-infobox').length) {
+					return 'portable-infobox-link';
+				} else if ($element.closest('.context-link').length) {
+					return 'context-link';
+				} else if ($element.closest('blockquote').length) {
+					return 'blockquote-link';
+				}
+				return 'regular-link';
```
Comment (tomasznapieralski) [inline front/main/app/components/article-content.js]: please add line above `return` statement

File: front/main/app/components/article-content.js [hunk 5]
```diff
@@ -176,6 +187,24 @@ export default Ember.Component.extend(
...
+			}
+			return '';
```
Comment (tomasznapieralski) [inline front/main/app/components/article-content.js]: please add line above `return` statement

File: front/main/app/components/article-content.js [hunk 6]
```diff
@@ -176,6 +187,24 @@ export default Ember.Component.extend(
 			}
 		},
 
+		/**
+		 * @param {jQuery} $element — element of which context is to be checked
+		 * @returns {string}
+		 */
+		getTrackingEventLabel($element) {
+			if ($element.length) {
+				if ($element.closest('.portable-infobox').length) {
```
Comment (tomasznapieralski) [inline front/main/app/components/article-content.js]: you don't need to check length, closest can return element or null

File: front/main/app/components/article-content.js [hunk 7]
```diff
@@ -176,6 +187,24 @@ export default Ember.Component.extend(
...
+					return 'portable-infobox-link';
+				} else if ($element.closest('.context-link').length) {
```
Comment (tomasznapieralski) [inline front/main/app/components/article-content.js]: you don't need to check length, closest can return element or null

File: front/main/app/components/article-content.js [hunk 8]
```diff
@@ -176,6 +187,24 @@ export default Ember.Component.extend(
...
+					return 'context-link';
+				} else if ($element.closest('blockquote').length) {
```
Comment (tomasznapieralski) [inline front/main/app/components/article-content.js]: you don't need to check length, closest can return element or null

File: front/main/app/components/article-content.js (hunk 6 above)
Comment (Warkot) [inline front/main/app/components/article-content.js]: I must disagree, `closest` always returns list of elements—if there are no elements found, it's just an empty list. And in JS `Boolean([]) === true`.

File: front/main/app/components/article-content.js (hunk 7 above)
Comment (Warkot) [inline front/main/app/components/article-content.js]: https://github.com/Wikia/mercury/pull/2112#discussion_r55197687

File: front/main/app/components/article-content.js (hunk 8 above)
Comment (Warkot) [inline front/main/app/components/article-content.js]: https://github.com/Wikia/mercury/pull/2112#discussion_r55197687

File: front/main/app/components/article-content.js (hunk 6 above)
Comment (vforge) [inline front/main/app/components/article-content.js]: @Warkot can you jsfiddle your solution?

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Pull request merged**